lib/adapters/http/read.js

```diff
@@ -1,13 +1,94 @@
+'use strict';
 var _ = require('underscore');
 var contentType = require('content-type');
-var Juttle = require('../../runtime').Juttle;
+var AdapterRead = require('../../runtime/adapter-read');
 var parsers = require('../parsers');
+var errors = require('../../errors');
 var request = require('request');
+var Promise = require('bluebird');
+
+class Iterator {
+    constructor(adapter, req, res) {
+        this.points = [];
+        this.done = false;
```

Comment:
I don't really see the point of splitting it into a separate `start` function since there's no case in which you would use an iterator without starting it immediately. Also FWIW I was burned for a while by the fact that if you don't start actually reading from the `res` object inside the `.on('response', (res) => {...}` handler, then you get no data at all.



However I do agree the constructor is long, so I split out the guts of the point processing into a separate function inside the iterator which tightens things up somewhat.